### PR TITLE
[python-nextgen] Add ApiResponse object

### PR DESCRIPTION
**Pull request by wing328**
Add ApiResponse object, which contains status code, headers, data (deserialized), raw data

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

FYI @taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @arun-nalla (2019/11) @spacether (2019/11) @krjakbrjak (2023/02)

### CODE REVIEW

**Comment by wing328** [comment]
circleci failure not related to this change.

File: modules/openapi-generator/src/main/resources/python-nextgen/api_response.mustache [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    status_code: Optional[StrictInt]
+    headers: Optional[Dict[StrictStr, StrictStr]]
+    data: Optional[Any]
+    raw_data: Optional[Any]
+
+    def __init__(self,
+                 status_code=None,
+                 headers=None,
+                 data=None,
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/python-nextgen/api_response.mustache:18]: What do this and raw_data store? Json? How about adding a more descriptive type hint or using generics?

File: modules/openapi-generator/src/main/resources/python-nextgen/api.mustache [hunk 2]
```diff
@@ -62,10 +63,6 @@ class {{classname}}(object):
 {{/allParams}}
         :param async_req: Whether to execute the request asynchronously.
         :type async_req: bool, optional
-        :param _preload_content: if False, the urllib3.HTTPResponse object will
```
Comment (tomplus) [inline modules/openapi-generator/src/main/resources/python-nextgen/api.mustache:65]: It's very useful switch, especially if you need data in a raw form for some reason (performance mainly). Can we keep this in some form?

Comment (spacether) [reply modules/openapi-generator/src/main/resources/python-nextgen/api.mustache:65]: Why not store the http raw response in ApiRsponse?

Comment (tomplus) [reply modules/openapi-generator/src/main/resources/python-nextgen/api.mustache:65]: Sure, ApiResponse.raw_data is the best place, but if _preload_content is False, deserialize() is not called and ApiResponse.data is None.

Comment (spacether) [reply modules/openapi-generator/src/main/resources/python-nextgen/api.mustache:65]: That's fine. One could keep raw_data as optional and add a comment describing that

Comment (wing328) [reply modules/openapi-generator/src/main/resources/python-nextgen/api.mustache:65]: @tomplus  The data in raw form (HTTP response body without deserialisation) is stored in `raw_data` in ApiResponse. Would that meet your requirement?

File: modules/openapi-generator/src/main/resources/python-nextgen/api_response.mustache (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/python-nextgen/api_response.mustache:20]: updated with description

File: modules/openapi-generator/src/main/resources/python-nextgen/api.mustache (hunk 2 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/python-nextgen/api.mustache:66]: UPDATE: added back the switch

Comment (tomplus) [reply modules/openapi-generator/src/main/resources/python-nextgen/api.mustache:66]: Looks good to me, thanks!

**Review by tomplus — APPROVED** [review]

**Pull request merged**